cocos/audio/android/PcmAudioPlayerPool.cpp

```diff
@@ -0,0 +1,130 @@
+        : _engineItf(engineItf)
+        , _outputMixObject(outputMixObject)
+        , _deviceSampleRate(deviceSampleRate)
+        , _deviceBufferSizeInFrames(deviceBufferSizeInFrames)
+{
+    _audioPlayerPool.reserve(AUDIO_PLAYER_POOL_SIZE);
+
+    for (int i = 0; i < AUDIO_PLAYER_POOL_SIZE; ++i)
+    {
+        int numChannels = 1;
+        if (i >= (AUDIO_PLAYER_POOL_SIZE / 2))
+        {
+            numChannels = 2;
+        }
+
+        auto player = createPlayer(numChannels);
+        if (player != nullptr)
+        {
+            LOGD("Insert a PcmAudioPlayer (%p, channels:%d) to pool ...", player, numChannels);
+            _audioPlayerPool.push_back(player);
+        }
+        else
+        {
+            LOGW("Could not create more PcmAudioPlayers, current count: %d", i);
+            break;
+        }
+    }
+}
+
+PcmAudioPlayerPool::~PcmAudioPlayerPool()
+{
+    LOGD("PcmAudioPlayerPool::destroy begin ...");
+    int i = 0;
+    for (const auto& player : _audioPlayerPool)
+    {
+        LOGD("delete PcmAudioPlayer (%d, %p) ...", i, player);
+        delete player;
+        ++i;
+    }
+    _audioPlayerPool.clear();
+    LOGD("PcmAudioPlayerPool::destroy end ...");
+}
+
+PcmAudioPlayer *PcmAudioPlayerPool::findAvailablePlayer(int numChannels)
+{
+    std::vector<PcmAudioPlayer*>::iterator freeIter = _audioPlayerPool.end();
+    int i = 0;
+    for (auto iter = _audioPlayerPool.begin(); iter != _audioPlayerPool.end(); ++iter)
+    {
+        if ((*iter)->getState() == IAudioPlayer::State::INITIALIZED)
+        {
+            if (freeIter == _audioPlayerPool.end())
+            {
+                freeIter = iter;
+            }
+
+            if ((*iter)->getChannelCount() == numChannels)
+            {
+                LOGD("PcmAudioPlayer (%p, idx: %d) is working ...", (*iter), i);
+                return (*iter);
+            }
+        }
+        ++i;
+    }
+
+    // Try to delete a free player and create a new (std::nothrow) one matches the channel count
+    if (freeIter != _audioPlayerPool.end())
+    {
+        LOGD("Removing a player (%p, channel:%d)", (*freeIter), (*freeIter)->getChannelCount());
+        delete (*freeIter);
+        _audioPlayerPool.erase(freeIter);
+        auto player = createPlayer(numChannels);
+        if (player != nullptr)
+        {
+            LOGD("Insert a PcmAudioPlayer (%p, channels:%d) to pool ...", player, numChannels);
+            _audioPlayerPool.push_back(player);
+            return player;
+        }
+    }
+
+    LOGE("Could not find available audio player with %d channels!", numChannels);
+    return nullptr;
+}
+
+PcmAudioPlayer *PcmAudioPlayerPool::createPlayer(int numChannels)
+{
+    PcmAudioPlayer* player = new (std::nothrow) PcmAudioPlayer(_engineItf, _outputMixObject);
+    if (player != nullptr)
+    {
+        if (!player->init(numChannels, _deviceSampleRate,
+                          _deviceBufferSizeInFrames * numChannels * 2))
```

Comment:
Hi, @j-jorge , please refer to `PcmAudioPlayer::init`, there are some `SL_RETURN_VAL_IF_FAILED` which will return false if `OpenSLES` function invocation fails.